docs/source/recipes/deploying-elyra-in-a-jupyterhub-endvironment.md

```diff
@@ -0,0 +1,119 @@
+<!--
+{% comment %}
+Copyright 2018-2020 IBM Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+# Deploying Elyra & JupyterHub in a Kubernetes environment
+
+This document goes over how to integrate Elyra into a self-service Jupyter Notebook platform
+managed by JupyterHub.
+
+If you are looking for details on how to install/configure JupyterHub, please look at
+[Zero to JupyterHub with Kubernetes](https://zero-to-jupyterhub.readthedocs.io/en/latest/index.html)
+or in this [Ansible scripts to deploy Elyra, JupyterHub, Jupyter Enterprise Gateway](https://github.com/lresende/ansible-kubernetes-cluster)
+
+![JupyterHub Deployment](https://raw.githubusercontent.com/lresende/ansible-kubernetes-cluster/master/docs/images/elyra-deployment-diagram.png)
+
+
+## Configuring JupyterHub Deployment
+
+JupyterHub is the entry point for our solution, it will manage user authorization and provisioning
+of individual Notebook servers for each user.
+
+JupyterHub configuration is done via a config.yaml, and the following settings are required:
+
+ * Configure JupyterHub to use Elyra docker image when instantiating the notebook environment for each user
+
+```yaml
+singleuser:
+  defaultUrl: "/lab"
+  image:
+    name: elyra/elyra
+    tag: dev
+    # disable this in a production environment
+    pullPolicy: "Always"
+  storage:
+    dynamic:
+      storageClass: nfs-dynamic
+```
+
+## Deploying Jupyter Enterprise Gateway
+
+Jupyter Enterprise Gateway enable support for remote kernels in a Jupyter Deployment
+
+ * Deploying Jupyter Enterprise gateway
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/jupyter/enterprise_gateway/master/etc/kubernetes/enterprise-gateway.yaml
```

Comment:
I'm not sure using `master` here is the right approach since this will pull the EG dev image, whatever that happens to be at the time.  Perhaps referencing a release version would be better?
https://raw.githubusercontent.com/jupyter/enterprise_gateway/v2.1.1/etc/kubernetes/enterprise-gateway.yaml